Dashboard/pages/add_product.py

```diff
@@ -0,0 +1,67 @@
+from os import _Environ, environ as CONFIG
+import logging
+
+import streamlit as st
+from psycopg2.extensions import connection
+
+from navigation import make_sidebar
+
+from helpers import (
+    get_connection, can_parse_as_float,
+    get_supported_websites,
+    update_db_to_subscribe
+)
+from load import load_product_data
+
+def subscribe_to_product(conn: connection, product_url: str, price_threshold: bool | None) -> bool:
+    """adds a product to the database and return True if successful
+    (or if product is already in the database), else False."""
+    try:
+        load_product_data(conn, product_url)
+    except Exception:
+        logging.error(
+            "Invalid URL, please enter a URL from one of the supported websites!")
+        st.warning(
+            "Invalid URL, please enter a URL from one of the supported websites!")
+        return False
+    try:
+        if price_threshold == "":
+            price_threshold = None
+        update_db_status = update_db_to_subscribe(conn, product_url,
+                                      price_threshold, st.session_state["email"])
+        if update_db_status['status']:
+            st.success(update_db_status['message'])
+        else:
+            st.warning(update_db_status['message'])
+    except Exception:
```

Comment:
more specific error please

Reply:
This is client facing, so preventing errors from reaching customers if necessary